styx-common/src/main/java/com/spotify/styx/state/FanOutputHandler.java

```diff
@@ -48,4 +52,17 @@ public void transitionInto(RunState state) {
       }
     }
   }
+
+  @Override
+  public void transitionInto(RunState state) {
+    transitionInto(state, outputHandlers);
+  }
+
+  @Override
+  public void tryTransitionInto(RunState state) {
+    final List<OutputHandler> outputHandlers = StreamSupport.stream(this.outputHandlers.spliterator(), false)
+        .filter(not(outputHandler -> outputHandler instanceof AtMostOnceOutputHandler))
```

Comment:
So the `FanOutputHandler` would not make forward any calls to `AtMostOnceOutputHandler` instances at all? That's surprising behavior.

Then there is nothing calling the `PublisherHandler`, unless you'd add some additional logic e.g. to the QueuedStateManager to specifically call the `AtMostOnceOutputHandler` instances. At which point this is starting to add more complexity than I'd want.

Reply:
Given tryTransitionInto is designed to be called outside of state machine, yes. But I agree, I don't like this very much either.